src/instructlab/model/serve.py

```diff
@@ -158,6 +164,17 @@ def serve(
         # Warn if unsupported backend parameters are passed
         warn_for_unsuported_backend_param(ctx)
 
+        if gpus:
+            # if `--tensor-parallel-size is included in ctx.args (click arguments after "--"),
+            # that value has precedence over the value of the --gpus flag.
+            if "--tensor-parallel-size" not in ctx.args:
```

Comment:
Should use contains_argument from vllm.  Generate does this now and evaluate should switch to use it at some point.